### ISSUE TITLE
WireGuard: Compile error with 4.13.x with debug kernel config

### ISSUE THREAD

**Issue by rn** [linuxkit/linuxkit#2669]
Using:
```
ENV WIREGUARD_VERSION=0.0.20171101
ENV WIREGUARD_SHA256=096b6482a65e566c7bf8c059f5ee6aadb2de565b04b6d810c685f1c377540325
```

Compiling the LinuxKit debug kernels, e.g., `make build_4.13.x-dbg` i get the error below. Tried with 4.13.11-dbg, 4.13.10-dbg and 4.9.59-dbg.

The debug kernels enable a bunch of extra options, see https://github.com/linuxkit/linuxkit/blob/master/kernel/config-dbg

I suspect 328e3861dcb034d38c7cc10007067abd3eebe39f may be the culprit.

/ping @zx2c4 

```
Removing intermediate container f674b18ff117
Step 20/31 : RUN curl -sSL -o /wireguard.tar.xz "${WIREGUARD_URL}" &&     echo "${WIREGUARD_SHA256}  /wireguard.tar.xz" | sha256sum -c - &&     tar -C / --one-top-level=wireguard --strip-components=2 -xJf /wireguard.tar.xz "WireGuard-${WIREGUARD_VERSION}/src" &&     make -j "$(getconf _NPROCESSORS_ONLN)" M="/wireguard" modules
 ---> Running in e5e4d1553e13
/wireguard.tar.xz: OK
getconf: LFS_CFLAGS: unknown variable
getconf: LFS_LDFLAGS: unknown variable
getconf: LFS_LIBS: unknown variable
  CC [M]  /wireguard/main.o
  CC [M]  /wireguard/noise.o
  CC [M]  /wireguard/device.o
  CC [M]  /wireguard/peer.o
  CC [M]  /wireguard/timers.o
  CC [M]  /wireguard/queueing.o
  CC [M]  /wireguard/send.o
  CC [M]  /wireguard/receive.o
  CC [M]  /wireguard/socket.o
  CC [M]  /wireguard/hashtables.o
  CC [M]  /wireguard/routingtable.o
  CC [M]  /wireguard/ratelimiter.o
  CC [M]  /wireguard/cookie.o
  CC [M]  /wireguard/netlink.o
  CC [M]  /wireguard/crypto/curve25519.o
  CC [M]  /wireguard/crypto/chacha20poly1305.o
  CC [M]  /wireguard/crypto/blake2s.o
  AS [M]  /wireguard/crypto/chacha20-ssse3-x86_64.o
  AS [M]  /wireguard/crypto/poly1305-sse2-x86_64.o
  AS [M]  /wireguard/crypto/blake2s-avx-x86_64.o
  AS [M]  /wireguard/crypto/curve25519-avx-x86_64.o
  AS [M]  /wireguard/crypto/chacha20-avx2-x86_64.o
  AS [M]  /wireguard/crypto/poly1305-avx2-x86_64.o
In file included from ./include/linux/rbtree.h:34:0,
                 from ./include/linux/mm_types.h:9,
                 from ./include/linux/kmemcheck.h:4,
                 from ./include/linux/skbuff.h:18,
                 from ./include/linux/ip.h:20,
                 from /wireguard/compat/compat.h:512,
                 from <command-line>:0:
/wireguard/noise.c: In function 'noise_keypair_get':
/wireguard/noise.c:121:56: error: expected ')' before '__func__'
  RCU_LOCKDEP_WARN(!rcu_read_lock_bh_held(), "Calling " __func__ " without holding the RCU BH read lock");
                                                        ^
./include/linux/rcupdate.h:289:47: note: in definition of macro 'RCU_LOCKDEP_WARN'
    lockdep_rcu_suspicious(__FILE__, __LINE__, s); \
                                               ^
In file included from ./include/linux/rbtree.h:34:0,
                 from ./include/linux/mm_types.h:9,
                 from ./include/linux/kmemcheck.h:4,
                 from ./include/linux/skbuff.h:18,
                 from ./include/linux/ip.h:20,
                 from /wireguard/compat/compat.h:512,
                 from <command-line>:0:
/wireguard/peer.c: In function 'peer_get':
/wireguard/peer.c:61:56: error: expected ')' before '__func__'
  RCU_LOCKDEP_WARN(!rcu_read_lock_bh_held(), "Calling " __func__ " without holding the RCU read lock");
                                                        ^
./include/linux/rcupdate.h:289:47: note: in definition of macro 'RCU_LOCKDEP_WARN'
    lockdep_rcu_suspicious(__FILE__, __LINE__, s); \
                                               ^
make[1]: *** [scripts/Makefile.build:303: /wireguard/peer.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [scripts/Makefile.build:303: /wireguard/noise.o] Error 1
make: *** [Makefile:1512: _module_/wireguard] Error 2
```

**Comment by zx2c4** [comment]
Fixed upstream here:

https://git.zx2c4.com/WireGuard/patch/?id=cc310a5dd1899417971e993d61d1230b14fe4f87

Sorry, checkpatch.pl did nonsense.

To get things working in the meantime, you can safely just do:

```bash
sed -i '/RCU_LOCKDEP_WARN/d' peer.c noise.c
```

**Comment by rn** [comment]
closing. This was fixed with #2740
